android/app/src/main/java/com/example/hajken/MainActivity.java

```diff
@@ -79,4 +81,9 @@ public void onBackPressed(){
             }
         }
     }
+
+    // hack to get activity passable in different fragments
+    public static MainActivity getThis(){
```

Comment:
Is this the generally suggested way to do this? I don't care too much if it works, but design-wise it looks slightly fishy (i.e. a circular dependency)! :grin: